data-manager/jdbc/src/test/java/org/orbisgis/orbisdata/datamanager/jdbc/JdbcDataSourceTest.java

```diff
@@ -867,6 +867,159 @@ void testGetDataSet() {
         assertTrue(dataset instanceof ITable);
     }
 
+    /**
+     * Test the {@link JdbcDataSource#getTableNames(String)},
+     * {@link JdbcDataSource#getTableNames(String, IJdbcDataSource.TableType...)},
+     * {@link JdbcDataSource#getTableNames(String, String)},
+     * {@link JdbcDataSource#getTableNames(String, String, IJdbcDataSource.TableType...)},
+     * {@link JdbcDataSource#getTableNames(String, String, String)},
+     * {@link JdbcDataSource#getTableNames(String, String, String, IJdbcDataSource.TableType...)} methods.
+     */
+    @Test
+    void getTableNameTest() {
+        assertTrue(ds1.getTableNames("toto").isEmpty());
+        assertTrue(ds2.getTableNames("toto").isEmpty());
+        assertFalse(ds1.getTableNames("TEST_H2GIS").isEmpty());
+        assertFalse(ds1.getTableNames(null).isEmpty());
+        assertFalse(ds2.getTableNames("test_postgis").isEmpty());
+        assertFalse(ds2.getTableNames(null).isEmpty());
+
+
+        assertTrue(ds1.getTableNames("TEST_H2GIS", UNKOWN, FOREIGN_TABLE, TABLE_LINK, TEMPORARY, VIEW).isEmpty());
+        assertFalse(ds1.getTableNames("TEST_H2GIS", TABLE).isEmpty());
+        assertTrue(ds1.getTableNames("TEST_H2GIS", UNKOWN, FOREIGN_TABLE, TABLE_LINK, TEMPORARY).isEmpty());
+        assertFalse(ds1.getTableNames("TEST_H2GIS", TABLE, VIEW).isEmpty());
+        assertFalse(ds1.getTableNames(null, TABLE).isEmpty());
+        assertFalse(ds1.getTableNames("TEST_H2GIS", (IJdbcDataSource.TableType) null).isEmpty());
+        assertFalse(ds1.getTableNames(null, (IJdbcDataSource.TableType) null).isEmpty());
+
+        assertTrue(ds2.getTableNames("test_postgis", UNKOWN, FOREIGN_TABLE, TABLE_LINK, TEMPORARY, VIEW).isEmpty());
+        assertFalse(ds2.getTableNames("test_postgis", TABLE).isEmpty());
+        assertTrue(ds1.getTableNames("TEST_H2GIS", UNKOWN, FOREIGN_TABLE, TABLE_LINK, TEMPORARY).isEmpty());
+        assertFalse(ds2.getTableNames("test_postgis", TABLE, VIEW).isEmpty());
+        assertFalse(ds2.getTableNames(null, TABLE).isEmpty());
+        assertFalse(ds2.getTableNames("test_postgis", (IJdbcDataSource.TableType) null).isEmpty());
+        assertFalse(ds2.getTableNames(null, (IJdbcDataSource.TableType) null).isEmpty());
```

Comment:
Not sure it's valuable to test ds2 because it's the same information schema no ?